### PR TITLE
HIVE-24021: Read insert-only tables truncated by Impala correctly

### PR DESCRIPTION
**Pull request by klcopp**
This is the second PR for this change.

### Why are the changes needed?
Hive reads insert-only tables truncated by Impala as if the truncation hadn't happened.

### Does this PR introduce _any_ user-facing change?
Yes, getAcidState will not filter out files with names beginning with "_empty"

### How was this patch tested?
Unit test

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java [hunk 1]
```diff
@@ -220,6 +220,10 @@ public boolean accept(Path p) {
       if (name.startsWith(OrcAcidVersion.ACID_FORMAT)) {
         return true;
       }
+      // Don't filter out empty files
+      if (name.startsWith("_empty")) {
```
Comment (pvargacl) [inline ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java:224]: extract to a constant, please.

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommandsForMmTable.java [hunk 2]
```diff
@@ -481,6 +482,61 @@ public void testOperationsOnCompletedTxnComponentsForMmTable() throws Exception
...
+
+    // rename empty file to "_empty"
+    Path basePath = new Path(tblLocation, "base_0000003");
+    Assert.assertTrue("Rename failed",
+        fs.rename(new Path(basePath, "000000_0"), new Path(basePath, "_empty")));
+
+    // 3. Verify query result. Selecting from a truncated table should return nothing.
+    List<String> rs = runStatementOnDriver("select a,b from " + TableExtended.MMTBL + " order by a,b");
+    Assert.assertEquals(Collections.emptyList(), rs);
+  }
```
Comment (pvargacl) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommandsForMmTable.java:538]: Could you extend the test to run a compaction and read back the data, just to make sure compaction also handles this file properly?

Comment (klcopp) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommandsForMmTable.java:536]: Great idea!

**Comment by pvargacl** [comment]
LGTM. +1

**Pull request closed without merging**